plugin/pxr/cmake/macros/Public.cmake

```diff
@@ -352,6 +352,12 @@ function(pxr_register_test TEST_NAME)
             endif()
         endif()
 
+        if(IS_MACOSX)
+            # Necessary for tests to find libraries, like python/tbb from Maya.
+            set(testWrapperCmd ${testWrapperCmd} --env-var=DYLD_LIBRARY_PATH=${MAYA_LOCATION}/Maya.app/Contents/MacOS:$ENV{DYLD_LIBRARY_PATH})
+            set(testWrapperCmd ${testWrapperCmd} --env-var=DYLD_FRAMEWORK_PATH=${MAYA_LOCATION}/Maya.app/Contents/Frameworks:$ENV{DYLD_FRAMEWORK_PATH})
+        endif()
```

Comment:
Copy setup code for the maya-usd tests from [here](https://github.com/Autodesk/maya-usd/blob/4d446967e2a150a4bfa8f8c8a16addc364da821c/cmake/test.cmake#L531).